Validation/CTPPS/plugins/CTPPSHepMCDistributionPlotter.cc

```diff
@@ -65,9 +76,23 @@ CTPPSHepMCDistributionPlotter::CTPPSHepMCDistributionPlotter(const edm::Paramete
 
 //----------------------------------------------------------------------------------------------------
 
+void CTPPSHepMCDistributionPlotter::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<std::string>("lhcInfoLabel", "")->setComment("label of the LHCInfo record");
+  desc.add<std::string>("lhcInfoPerLSLabel", "")->setComment("label of the LHCInfoPerLS record");
+  desc.add<std::string>("lhcInfoPerFillLabel", "")->setComment("label of the LHCInfoPerFill record");
+  desc.add<bool>("useNewLHCInfo", false)->setComment("flag whether to use new LHCInfoPer* records or old LHCInfo");
+
+  desc.add<std::string>("outputFile", "")->setComment("output file");
+
+  descriptions.add("ctppsHepMCDistributionPlotter", desc);
+}
+
 void CTPPSHepMCDistributionPlotter::analyze(const edm::Event &iEvent, const edm::EventSetup &iSetup) {
   // get conditions
-  const auto &lhcInfo = iSetup.getData(lhcInfoESToken_);
+  LHCInfoCombined lhcInfoCombined(
+      iSetup, lhcInfoPerLSToken_, lhcInfoPerFillToken_, lhcInfoToken_, useNewLHCInfo_, iEvent.isRealData());
```

Comment:
This module is plotting HepMC stuff, so it's clearly running on MC. It won't even run on top of real data, so we can hard-code the usage of the old LHCInfo

Reply:
Well it didn't cost me much time to change it and now it offers some extra stuff like `isCrossingAngleValid` etc and maybe - that's actually something I don't know - `Direct Simulation` will one day be using new `LHCInfoPer*` records?

Reply:
I don't think so, because the simulation should be setup via manual configurations. It's hard to say if run-dependent simulation will ever be used for PPS, but I'd exclude it as an option for now